libretroshare/src/libretroshare.pro

```diff
@@ -861,45 +861,69 @@ rs_jsonapi {
     WRAPPERS_REG_FILE=$$clean_path($${JSONAPI_GENERATOR_OUT}/jsonapi-wrappers.inl)
 
     no_rs_cross_compiling {
-        restbed.target = $$clean_path($${RESTBED_BUILD_PATH}/library/librestbed.a)
-        restbed.commands = \
-            cd $${RS_SRC_PATH};\
-            git submodule update --init --recommend-shallow supportlibs/restbed;\
-            cd $${RESTBED_SRC_PATH};\
-            git submodule update --init --recommend-shallow dependency/asio;\
-            git submodule update --init --recommend-shallow dependency/catch;\
-            git submodule update --init --recommend-shallow dependency/kashmir;\
-            mkdir -p $${RESTBED_BUILD_PATH}; cd $${RESTBED_BUILD_PATH};\
-            cmake -DCMAKE_CXX_COMPILER=$$QMAKE_CXX -DBUILD_SSL=OFF \
-                -DCMAKE_INSTALL_PREFIX=. -B. -H$$shell_path($${RESTBED_SRC_PATH});\
-            make; make install
-        QMAKE_EXTRA_TARGETS += restbed
-        libretroshare.depends += restbed
-        PRE_TARGETDEPS *= $${restbed.target}
+        DUMMYRESTBEDINPUT = FORCE
+        genrestbedlib.name = Generating libresbed.
+        genrestbedlib.input = DUMMYRESTBEDINPUT
+        genrestbedlib.output = $$clean_path($${RESTBED_BUILD_PATH}/librestbed.a)
+        genrestbedlib.CONFIG += target_predeps combine
+        genrestbedlib.variable_out = PRE_TARGETDEPS
+        win32-g++ {
```

Comment:
Do I miss something or the only difference here between the ` win32-g++` and `else` branch is the usage of `&&` instead of `;` in that case it seems reasonable to me (that is the original writer of that code) to use `&&` for all systems removing the conditional

Reply:
It makes sense to use && on Linux too, because ; ignores errors in previous steps.

But the significant change is this cmake option, else cmake will try to use VisualC:
-G \"MSYS Makefiles\"

Reply:
So only that part should be generated in an os specific branch, you can define a variable like 
```
CMAKE_GENERATOR_OVERRIDE=""
win32-g++:CMAKE_GENERATOR_OVERRIDE="-G \"MSYS Makefiles\""
```

and the use that in the `cmake` command line

`cmake -DCMAKE_CXX_COMPILER=$$QMAKE_CXX  $${CMAKE_GENERATOR_OVERRIDE} -DBUILD_SSL=OFF \ `